### DISCUSSION TITLE
Get all tags related to taggable class

### DISCUSSION THREAD

**Discussion by reesmcivor** [spatie/laravel-tags#415] [Ideas]
Hi there, 

I can't find this currently in any discussions. Apologies if it has been covered before.

I'm creating a search type functionality that includes listing all the tags in a select dropdown field but I couldn't find a method to get all tags using a specific taggable model. So I have come up with the following:

`Tag::query()->join('taggables', 'taggables.tag_id', '=', 'tags.id')->where('taggable_type', '=', Exercise::class)>get();`

It works fine but I just want to be sure there isn't anything out there already that I am missing from the current package.

**Comment by dangnhdev** [comment]
I'm looking for this too. Did you find any better way to do this?

**Comment by uyab** [comment]
In my case, what I need is to get tags that have relation with any model. Here is what I do:

1. create a custom `Tag` model
2. add inversed relation between Tag and model using `morphedByMany`
3. Get tags with `Tag::query()->whereHas('courses')->get()`

```php
class Tag extends \Spatie\Tags\Tag
{
    use HasUuids;

    public function courses(): \Illuminate\Database\Eloquent\Relations\MorphToMany
    {
        return $this->morphedByMany(Course::class, 'taggable');
    }
}
```
